workflow/controller/controller.go

```diff
@@ -753,10 +830,20 @@ func (wfc *WorkflowController) getMetricsServerConfig() (metrics.ServerConfig, m
 		Port:         port,
 		IgnoreErrors: wfc.Config.TelemetryConfig.IgnoreErrors,
 	}
-
 	return metricsConfig, telemetryConfig
 }
 
+func (wfc *WorkflowController) releaseAllWorkflowLocks(obj interface{}) {
+	un, ok := obj.(*unstructured.Unstructured)
+	if !ok {
+		log.Warnf("Key '%s' in index is not an unstructured", obj)
+	}
+	wf, err := util.FromUnstructured(un)
+	if err != nil {
+		log.Warnf("Invalid workflow object: %v", obj)
+	}
+	wfc.syncManager.ReleaseAll(wf)
+}
```

Comment:
Blank line below this function
```suggestion
}

```